components/grid/virtual-scrolling.md

```diff
@@ -79,54 +87,250 @@ To enable virtual scrolling:
+    }
+
+    .small-row-height.k-grid td {
+        margin: 0;
+        padding: 0;
+    }
+</style>
+
+<TelerikGrid OnRead="@OnGridRead"
+             TItem="@Product"
+             ScrollMode="@GridScrollMode.Virtual"
+             Height="480px" RowHeight="30" PageSize="20"
+             Sortable="true" FilterMode="@GridFilterMode.FilterRow"
+             Class="small-row-height">
+    <GridColumns>
+        <GridColumn Field="@nameof(Product.Name)" Title="Product Name" />
+        <GridColumn Field="@nameof(Product.Stock)" />
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private List<Product> GridData { get; set; } = new List<Product>();
+
+    private async Task OnGridRead(GridReadEventArgs args)
+    {
+        await Task.Delay(200); // simulate network delay
+
+        DataSourceResult result = GridData.ToDataSourceResult(args.Request);
+
+        args.Data = result.Data;
+        args.Total = result.Total;
+        args.AggregateResults = result.AggregateResults;
+    }
+
+    protected override void OnInitialized()
+    {
+        GridData = new List<Product>();
+        var rnd = new Random();
+
+        for (int i = 1; i <= 1000; i++)
+        {
+            GridData.Add(new Product()
+                {
+                    Id = i,
+                    Name = $"Product {i}",
+                    Stock = rnd.Next(0, 100)
+                });
+        }
+    }
+
+    public class Product
+    {
+        public int Id { get; set; }
+        public string Name { get; set; } = string.Empty;
+        public int Stock { get; set; }
+    }
+}
+````
+
+## Setting a Value for the PageSize Parameter
+
+Set the `PageSize` parameter to an `int` value. The `PageSize` determines how many rows are rendered at any given time and how many items are requested from the data source when loading data on demand. For optimal performance, use a page size that fills the grid's data viewport without being excessively large.
+
+## Considerations
+
+Virtualization primarily enhances client-side rendering performance and improves the user experience. However, it comes with the trade-off that certain features of the grid are not compatible with it. An alternative approach is to utilize [regular paging]({%slug components/grid/features/paging%}) combined with [manual data source operations]({%slug components/grid/manual-operations%}) to achieve the desired data retrieval performance.
+
+List of the known limitations of the virtual scrolling feature:
+
+* [Hierarchy]({%slug components/grid/features/hierarchy%}) is not supported.
+
+* [Grouping]({%slug components/grid/features/grouping%}) is not supported. [Loading Group Data On Demand]({%slug grid-group-lod%}) is supported, however.
 
 
 ## See Also
 
   * [Live Demo: Grid Virtual Scrolling](https://demos.telerik.com/blazor-ui/grid/virtual-scrolling)
+  * [Documentation: Selection in Grid with virtualized rows]({%slug components/grid/selection/overview%}#selection-in-grid-with-virtualized-rows)
```

Comment:
* Use title case capitalization as in the first list item.
* As we are in the docs, you don't need to call this out.
```suggestion
  * [Selection in Grid with Virtualized Rows]({%slug components/grid/selection/overview%}#selection-in-grid-with-virtualized-rows)
```